mllib/src/main/scala/org/apache/spark/mllib/linalg/CholeskyDecomposition.scala

```diff
@@ -52,8 +51,20 @@ private[spark] object CholeskyDecomposition {
   def inverse(UAi: Array[Double], k: Int): Array[Double] = {
     val info = new intW(0)
     lapack.dpptri("U", k, UAi, info)
-    val code = info.`val`
-    assert(code == 0, s"lapack.dpptri returned $code.")
+    checkReturnValue(info, "dpptri")
     UAi
   }
+
+  private def checkReturnValue(info: intW, method: String): Unit = {
```

Comment:
CC @yanboliang -- tried to add more to the error messages here, and distinguish between unsupported input (illegal argument) and bad call to LAPACK (illegal state? an assertion error seemed like too much)